apx/common/src/apx_nodeManager.c

```diff
@@ -739,7 +773,7 @@ static bool apx_nodeManager_setNoneDirtyInDataFromDefinition(apx_node_t *node, a
          {
             apx_node_fillPortInitData(node, port, portData);
             assert(packLen == (int32_t)adt_bytearray_length(portData));
-            memcpy(pNext, adt_bytearray_data(portData), packLen);
+            apx_nodeData_writeInPortDefaultDataIfNotDirty(nodeData, adt_bytearray_data(portData), offset, packLen);
```

Comment:
This is confusing, when you call this function you are checking if the flag is 0. But that has already been checked in the if-statement before this block. However, you seem to check the flags here without taking any lock.
The function apx_nodeData_writeInPortDefaultDataIfNotDirty at least takes a spinlock before flag check which seems appropriate.

Reply:
in the if-statement before this block there is a "quick and dirty" optimization check of the flag. before writing the port with content from the definition file there is validation that the port has not been written in the meantime -  this is the real check of the flag and spinlock is thus held.